packages/url_launcher/url_launcher_ios/pigeons/messages.dart

```diff
@@ -9,22 +9,55 @@ import 'package:pigeon/pigeon.dart';
   swiftOut: 'ios/Classes/messages.g.swift',
   copyrightHeader: 'pigeons/copyright.txt',
 ))
+
+/// Possible outcomes of launching a URL.
+enum LaunchResult {
+  /// The URL was successfully launched.
+  success,
+
+  /// The URL could not be launched
+  failure,
+
+  /// The URL was not launched because it is not invalid URL
+  invalidUrl,
+
+  /// The URL did not load successfully in the SFSafariViewController.
+  failedToLoad,
+}
+
+class LaunchResultDetails {
```

Comment:
I don't think we need this class; there isn't any information I see in the native error messages that comes from the native side, so we can just put those strings in the Dart code.